active/NumberBox/NumberBox.md

```diff
@@ -0,0 +1,184 @@
+
+## Examples
+<!-- Use this section to explain the features of the API, showing
+example code with each description. The general format is: 
+  feature explanation,
+  example code
+  feature explanation,
+  example code
+  etc.-->
+  
+<!-- Code samples should be in C# and/or C++/WinRT -->
+
+<!-- As an example of this section, see the Examples section for the PasswordBox control 
+(https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). -->
+
+### Create a NumberBox
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="simple NumberBox"/>
+```
+
+![](images/numberbox-basic.png)
+
+### Add placeholder text
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="simple NumberBox" Header="Enter equation:" PlaceholderText="A + B * C" />
+```
+
+![](images/numberbox-placeholder-text.png)
+
+### Enable calculation support
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="NumberBox for simple calculations" 
+    Header="Enter equation:" 
+    PlaceholderText="A + B * C" 
+    AcceptsCalculation="True" />
+```
+
+![](images/numberbox-acceptscalculation.png)
+
+### Add increment and decrement stepping
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="NumberBox for items in order" 
+    StepSize="0.1"
+    UpDownButtonsEnabled="True"
+    UpDownDragEnabled="True" />
+```
+
+![](images/numberbox-updownbuttons.png)
+
+### Add mininum and maximum values
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="NumberBox for items in order" 
+    PlaceholderText="1" 
+    MinValue="1"
+    MaxValue="99" />
+```
+
+## Remarks
+<!-- Explanation and guidance that doesn't fit into the Examples section. -->
+
+<!-- APIs should only throw exceptions in exceptional conditions; basically,
+only when there's a bug in the caller, such as argument exception.  But if for some
+reason it's necessary for a caller to catch an exception from an API, call that
+out with an explanation either here or in the Examples -->
+
+## API Notes
+<!-- Option 1: Give a one or two line description of each API (type
+and member), or at least the ones that aren't obvious
+from their name.  These descriptions are what show up
+in IntelliSense. For properties, specify the default value of the property if it
+isn't the type's default (for example an int-typed property that doesn't default to zero.) -->
+
+<!-- Option 2: Put these descriptions in the below API Details section,
+with a "///" comment above the member or type. -->
+
+## API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+
+```c++ 
+unsealed runtimeclass NumberBox : Windows.UI.Xaml.Controls.TextBox
+{
+    NumberBox();
+    
+    Double Value;
+    
+    Boolean AcceptsCalculation;
+    Boolean UpDownButtonsEnabled;
+    Boolean HyperDragEnabled;
```

Comment:
What are these hyper properties?

Reply:
Drag **and** scroll to step the same way up/down keyboard arrows or up/down buttons would. Adobe and Blender have set a delightful precedent that our community has advocated for on the proposal. I'll give @ryandemopoulos credit for the name.

Reply:
@SavoySchuler, I'm happy to help you work through the design details of all this dragging and scrolling. I think we'll need to add more detail about this behavior to the spec.

Reply:
A few details based on my implementation
1. Because dragging on a `TextBox` selects text in one - in [my implementation](https://github.com/xyzzer/WinRTXamlToolkit/blob/master/WinRTXamlToolkit/Controls/NumericUpDown/NumericUpDown.cs) I chose to use a transparent but hit-testable drag overlay element that captures manipulation events when the control doesn't have focus.
2. Tapping the drag overlay makes the control (the TextBox in its template) get focus
3. As long as the `TextBox` in the template has focus - the drag overlay hit test visibility is off.
4. The mouse cursor is hidden while dragging (`Window.Current.CoreWindow.PointerCursor = null;`), which enables continuous dragging, unbounded by screen borders
5. You might want to expose a property like `DragSpeed` since `StepSize` might not necessarily match how you'd want drag input to behave.
6. You might want your dragging speed to depend on screen resolution. Not sure why this worked for me better 6 years ago. Perhaps something changed - maybe this worked for quicker hyperdragging with touch on a smaller-screen Windows phone. Perhaps on a Surface Studio or Hub - this would actually be more painful today. Perhaps calculating physical offsets based on screen DPI would make more sense today.

I have an `IsReadOnly` property that you can use to block value updates using any means. If you add that - you need to make sure the `TextBox` has a matching value, but also that up/down keys/buttons and hyperdrag don't update the value.

Reply:
Tagging @KevinTXY for implementation details. 

With regard to `DragSpeed` I envisioned this not as a 1:1 to `StepSize`, but rather exponentially increasing over distance (and relative to Maximum value/screen size if a Maximum value is given) - however, my priority is to align with WinRT XAML Toolkit's NumericUpDown & Blender if able. Ultimately my goal is to avoid adding an implementation requirement for the developer, particularly in the case where drag speed may not be a static value. 

 @xyzzer, with this in mind, is this still a value you believe should have writable property exposed?

Reply:
HyperDrag hasn't been written and is somewhat an afterthought for me just because of it's complexity - I think I actually need to see how this works on other products first because I have doubts about it in general. By dragging are we emulating slider behavior or giving the box's value some kind of stepping velocity? How does this behave with decimal precision? Who is this for?